Workaround for bsc#1244437

badboywj opened the pull request:
cluster slave node didn't reboot after triggering network issue with iptables on SLE 16.0, here is the failed job:
https://openqa.suse.de/tests/17976099#step/priority_fencing_delay/16

create workaround for the issue after testing many times with cluster developer (Xin Liang).
increate timeout for assert_script_run because of the [failure](https://openqa.suse.de/tests/18039413#step/priority_fencing_delay/42)

- Related Bug: https://bugzilla.suse.com/show_bug.cgi?id=1244437
- Related MR: https://gitlab.suse.de/qa-css/openqa_ha_sap/-/merge_requests/1030
- Needles: None
- Verification run: 
    https://openqa.suse.de/tests/18093416 (x86_64: node01)
    https://openqa.suse.de/tests/18093414 (x86_64: node02)
    https://openqa.suse.de/tests/18116697 (ppc64le: node01)
    https://openqa.suse.de/tests/18116698 (ppc64le: node02)

alvarocarvajald reviewed (approved):
Code LGTM; however I wonder whether we should add a:
```
record_soft_failure 'bsc#1244437 - Priority Fencing Delay configuration requires a cluster restart in 16';
```
inside the if-block to at least track for the time being, until we're certain that this will be documented in the release notes or other documentation.

lpalovsky reviewed (approved):
LGTM

alvarocarvajald commented on tests/ha/priority_fencing_delay.pm:
```diff
@@ -60,6 +60,11 @@ sub run {
         assert_script_run "crm configure property priority-fencing-delay=30";
+        # Workaround for bsc#1244437
+        if (is_sle('>=16')) {
+            assert_script_run "crm cluster restart --all";
```
After seeing https://bugzilla.suse.com/show_bug.cgi?id=1244437#c6, we do need a `record_soft_failure` here.

badboywj replied:
I updated the code.

badboywj commented:
> Basically scenario fails. With priority fencing configured and a split brain situation force in L77 of the same module, we expect the non-prioritized node to be fenced and this is not happening. Work around to make the test pass is OK, but as you and @mpagot say, let's add a soft failure and perhaps a switch so we know what we're testing and not forget.

thank you Alvaro, good suggestion. I updated the code and added soft failure and a switch, and then I will update the test settings later.

badboywj commented:
> Developer said "Run `crm cluster restart --all` is just a workaround, not the last solution". If this issue is fixed, we need to remove this workaround to confirm that, how about adding a switch for enabling this workaround?
>
 
thank you, I added a switch for this workaround

> And I also have a question why do we need the workaround? Since this is the last test point in test case (`check_logs` is also tested in other jobs).

this issue is not related to this test, maybe it is better to know if the test can pass on SLE 16.0, so I created a workaround.

mpagot commented on tests/ha/priority_fencing_delay.pm:
```diff
@@ -62,6 +62,15 @@ sub run {
     }
 
+    # Workaround for bsc#1244437
+    if (is_node(2)) {
```
```suggestion
    if (is_node(2) && get_var("WORKAROUND_BSC1244437")) {
```

alvarocarvajald commented on tests/ha/priority_fencing_delay.pm:
```diff
@@ -13,7 +13,8 @@ use warnings;
 use hacluster;
-use utils qw(zypper_call);
+use utils qw(zypper_call reconnect_mgmt_console);
+use Utils::Backends 'is_pvm';
```
Not a big deal, but better use `qw(is_pvm)` instead of `'is_pvm'`. Makes adding more functions from that library easier, and it's more uniform with the other `use` lines in this module.

badboywj replied:
good suggestion, and updated the code.

alvarocarvajald reviewed (approved):
LGTM, but added a recommendation regarding functions imports.

lilyeyes reviewed (approved):
LGTM.
Only have a minor concern: does this PR work for other product 12-SP5?

badboywj commented:
> LGTM. Only have a minor concern: does this PR work for other product 12-SP5?

No, here is the comment from developer:
https://bugzilla.suse.com/show_bug.cgi?id=1244437#c4

but I think maybe it works on 12-SP5 with the switch `WORKAROUND_BSC1244437` once the same issue happens

mpagot commented on tests/ha/priority_fencing_delay.pm:
```diff
@@ -60,6 +63,14 @@ sub run {
         assert_script_run "crm configure property priority-fencing-delay=30";
+        # Workaround for bsc#1244437
+        if (get_var("WORKAROUND_BSC1244437")) {
+            assert_script_run "crm configure location force_primary promotable-1 100: $hostname";
```
```suggestion
            assert_script_run 'crm configure location force_primary promotable-1 100: ' . get_hostname;
```

as only used once, you do not need a variable.

badboywj replied:
Updated the code :+1:

alvarocarvajald reviewed (approved):
LGTM

alvarocarvajald commented:
> > LGTM. Only have a minor concern: does this PR work for other product 12-SP5?
> 
> No, here is the comment from developer: https://bugzilla.suse.com/show_bug.cgi?id=1244437#c4
> 
> but I think maybe it works on 12-SP5 with the switch `WORKAROUND_BSC1244437` once the same issue happens

This should be only for 16, right? Perhaps you can add a `is_sle('16+')` in that conditional.
